framework/include/postprocessors/SideAdvectiveFluxIntegral.h

```diff
@@ -0,0 +1,60 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// MOOSE includes
+#include "SideIntegralPostprocessor.h"
+#include "MooseVariableInterface.h"
+
+// Forward Declarations
+template <bool, typename>
+class SideAdvectiveFluxIntegralTempl;
+typedef SideAdvectiveFluxIntegralTempl<false, Real> SideAdvectiveFluxIntegral;
+typedef SideAdvectiveFluxIntegralTempl<true, Real> ADSideAdvectiveFluxIntegral;
+
+/**
+ * This postprocessor computes a side integral of the mass flux.
+ */
+template <bool is_ad, typename T>
```

Comment:
why is this templated? you use neither template parameters

you need to template for material properties, but you are using functor material properties right now

Reply:
I have switched to normal material properties and template

Reply:
why the T template? 
Are we planning to advect RealVectorValue instead of Real properties?